iModelCore/ECDb/ECDb/ECSql/PragmaECSqlPreparedStatement.cpp

```diff
@@ -0,0 +1,1092 @@
+        if (mapInfo.IsNotMapped() || mapInfo.IsEmpty()) {
+            ecdb.GetImpl().Issues().ReportV(
+                IssueSeverity::Error,
+                IssueCategory::BusinessProperties,
+                IssueType::ECSQL, "PRAGMA %s, is not invoked on class %s which is not mapped or exist.", GetName().c_str(), cls.GetFullName());
+            return BE_SQLITE_ERROR;
+        }
+        if (val.GetBool()) {
+            m_disqualifiedClassSet.insert(cls.GetId());
+        } else {
+            m_disqualifiedClassSet.erase(cls.GetId());
+        }
+        rowSet = std::make_unique<StaticPragmaResult>(ecdb);
+        rowSet->FreezeSchemaChanges();
+        return BE_SQLITE_OK;
+    }
+    static std::unique_ptr<PragmaManager::Handler> Create () { return std::make_unique<DisqualifyTypeIndex>(); }
+};
+
+//================================================================================
+// @bsiclass PragmaECDbVersion
+//================================================================================
+struct PragmaECDbVersion : PragmaManager::GlobalHandler {
+    PragmaECDbVersion():GlobalHandler("ecdb_ver","return current and file profile versions"){}
+    ~PragmaECDbVersion(){}
+    virtual DbResult Read(PragmaManager::RowSet& rowSet, ECDbCR ecdb, PragmaVal const&)  override {
+        auto result = std::make_unique<StaticPragmaResult>(ecdb);
+        result->AppendProperty("current", PRIMITIVETYPE_String);
+        result->AppendProperty("file", PRIMITIVETYPE_String);
+        result->FreezeSchemaChanges();
+        auto row = result->AppendRow();
+        row.appendValue() = ECDb::CurrentECDbProfileVersion().ToString();
+        row.appendValue() =  ecdb.GetECDbProfileVersion().ToString();
+        rowSet = std::move(result);
+        return BE_SQLITE_OK;
+    }
+    virtual DbResult Write(PragmaManager::RowSet& rowSet, ECDbCR ecdb, PragmaVal const&) override {
+        ecdb.GetImpl().Issues().ReportV(IssueSeverity::Error, IssueCategory::BusinessProperties, IssueType::ECSQL, "PRAGMA %s is readonly.", GetName().c_str());
+        rowSet = std::make_unique<StaticPragmaResult>(ecdb);
+        rowSet->FreezeSchemaChanges();
+        return BE_SQLITE_READONLY;
+    }
+    static std::unique_ptr<PragmaManager::Handler> Create () { return std::make_unique<PragmaECDbVersion>(); }
+};
+
+//================================================================================
+// @bsiclass PragmaFileInfo
+//================================================================================
+struct PragmaFileInfo : PragmaManager::GlobalHandler {
+    PragmaFileInfo():GlobalHandler("file_info","return file info"){}
+    ~PragmaFileInfo(){}
+    void AppendDbGuid(ECDbCR ecdb, BeJsValue row) {
+        row.appendValue() = "db_guid";
+        row.appendValue() =  ecdb.GetDbGuid().ToString();
+    }
+    void AppendProjectId(ECDbCR ecdb, BeJsValue row) {
+        row.appendValue() = "project_id";
```

Comment:
I don't particularly like exposing this as project id because it is not necessarily a projects id but I'm not sure we have a better name that fits in this scope.  iTwin id will be correct when used in the context of iTwin platform but all we can really say at the ECDb level is that this is the id of the larger context that the file belongs to.  I would suggest 'context' but we do not use 'context' else where in our API.